core/src/main/java/org/alfresco/util/TempFileProvider.java

```diff
@@ -518,32 +505,24 @@ private static int removeFiles(File directory, long removeBefore, long longLifeB
                     try
                     {
                         // only delete if the limits allow
-                        if (maxFilesToDelete != null && maxFilesToDelete.get() <= 0 ||
-                            maxTimeToRun != null && ((jobStartTime + maxTimeToRun.toMillis()) < System.currentTimeMillis()))
+                        if (shouldTheDeletionContinue())
```

Comment:
The name of the method is a bit confusing to me. 

>  if shouldTheDeletionContinue() is true then return count

I suggest renaming it to shouldTheDeleteStop() or something similar.

Reply:
That's correct! Thanks.